types/wlr_layer_shell_v1.c

```diff
@@ -88,6 +95,13 @@ static void layer_surface_handle_ack_configure(struct wl_client *client,
 static void layer_surface_handle_set_size(struct wl_client *client,
 		struct wl_resource *resource, uint32_t width, uint32_t height) {
 	struct wlr_layer_surface_v1 *surface = layer_surface_from_resource(resource);
+
+	if (!surface) {
+		wl_resource_post_error(resource,
```

Comment:
Hmm. I'm not sure invalid_surface_state has been designed for this. Maybe the layer surface should just become inert?

Reply:
that makes sense to me too. I think we had all the parts for that except for the `NULL` checks.